Adds basic implementation of Spinner component

procload opened the pull request:
## New Behavior
This PR creates a web component version of the Fluent Spinner component.

fabricteam commented:
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/be3d30fcbe222be34b02a554e948d14bb2d730df'>be3d30fcbe222be34b02a554e948d14bb2d730df</a></sub>
<div id="bundleSizeComment9423"/>

chrisdholt commented on packages/web-components/src/spinner/spinner.options.ts:
```diff
@@ -0,0 +1,43 @@
+   */
+  small: 'small',
+  tiny: 'tiny',
+  'extra-small': 'extra-small',
```
I'd suggest that we avoid keys as strings because the implementation of this is going to create an inconsistent experience. The important thing is that the _value_ is consistent for our types. In terms of keys, I'd suggest we follow our property convention of `camelCase`. With a string like this, in order to use it a developer is going to need to do:

```ts
const foo: SpinnerSize = SpinnerSize['extra-small'];
```

What I think we'll want (and what will be consistent between exported instances in FAST is:
```ts
const foo SpinnerSize = SpinnerSize.extraSmall;
```
This will give us a consistent application and use across. Similarly, when we hit numbers, we tend to prefix the actual key with an underscore. So if we look at Text size, you'll notice that the value is consistent but due to constraints w/ TS we can't use a number as a key. So to consistently use dot notation, we would have `TextSize._100` rather than TextSize['100']. Example below of this consideration as well:

<img width="323" alt="image" src="https://user-images.githubusercontent.com/13071055/214166582-9b13a22f-ed09-4f4f-a4ed-9c18af680a3f.png">

chrisdholt commented on packages/web-components/src/spinner/spinner.options.ts:
```diff
@@ -0,0 +1,43 @@
+  inverted: 'inverted',
+} as const;
+
+/**
+ * The types for Spinner size and appearance options
+ * @public
+ */
+export type SpinnerSize = Values<typeof SpinnerSize>;
+export type SpinnerAppearance = Values<typeof SpinnerAppearance>;
```
These should be separate with their own type docs.

chrisdholt reviewed (changes requested):
Just a couple things here - one last thing to remember is to create an export path for the component in the package.json so we can import just that registered component from the package.

procload commented:
> Just a couple things here - one last thing to remember is to create an export path for the component in the package.json so we can import just that registered component from the package.

Excellent, thank you! I updated those files and added it to the package.json. Once @miroslavstastny gives the go-ahead we can consider Spinner finished.

miroslavstastny commented on packages/web-components/src/spinner/spinner.options.ts:
```diff
@@ -0,0 +1,48 @@
+   * A Spinner's size can be small, tiny, extra-small, medium, large, extra-large, or huge
+   *
+   */
+  small: 'small',
+  tiny: 'tiny',
+  extraSmall: 'extra-small',
```
nit: can we keep the options sorted from the smallest to the biggest?
```suggestion
  tiny: 'tiny',
  extraSmall: 'extra-small',
  small: 'small',
```

miroslavstastny commented on packages/web-components/src/spinner/spinner.options.ts:
```diff
@@ -0,0 +1,48 @@
+ * The type for Spinner size option
+ * @public
+ */
+export type SpinnerSize = Values<typeof SpinnerSize>;
```
reorder: move this just bellow the `SpinnerSize` object to match the order used in another components

miroslavstastny replied:
Sorry, I was not clear enough.
The order should be:

```ts
export const SpinnerAppearance = {};
export type SpinnerAppearance = Values<typeof SpinnerAppearance>;

export const SpinnerSize = {};
export type SpinnerSize = Values<typeof SpinnerSize>;
```

miroslavstastny commented on packages/web-components/src/spinner/spinner.ts:
```diff
@@ -0,0 +1,28 @@
+   * The size of the spinner
+   *
+   * @public
+   * @remarks
```
```suggestion
   * @default 'medium'
   * @remarks
```

miroslavstastny commented on packages/web-components/src/spinner/spinner.ts:
```diff
@@ -0,0 +1,28 @@
+  /**
+   * The appearance of the progress bar
+   * @public
+   * @remarks
```
```suggestion
   * @default 'primary'
   * @remarks
```

miroslavstastny commented on packages/web-components/src/spinner/spinner.stories.ts:
```diff
@@ -0,0 +1,53 @@
+`;
+
+export default {
+  title: 'Web Components/Spinner',
```
```suggestion
  title: 'Components/Spinner',
```
![image](https://user-images.githubusercontent.com/9615899/214926763-5909e9da-7cfc-4142-ae8e-163f60be5b47.png)

procload commented:
Thanks, @miroslavstastny and @chrisdholt. Feedback addressed; this should be good to go!

chrisdholt commented on packages/web-components/src/spinner/spinner.options.ts:
```diff
@@ -0,0 +1,35 @@
+export type Values<T> = T[keyof T];
```
@procload we should have access to the ValuesOf helper which was in the last published version of Foundation. I think the other option files are using it. Can we remove this and use that?

procload replied:
Got it! Thanks.

chrisdholt commented on packages/web-components/src/spinner/spinner.options.ts:
```diff
@@ -0,0 +1,37 @@
+import { ValuesOf } from '@microsoft/fast-foundation';
+
+export type Values<T> = T[keyof T];
```
@procload I still see this, can we remove is it's duplicative of the above import?

```suggestion
```

procload replied:
Doh! Will do.

procload commented:
/azp run

procload commented:
/azp run
